src/lib/pubkey/kyber/kyber_common/kyber_helpers.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Crystals Kyber Internal Helpers
+ *
+ * Further changes
+ * (C) 2024 Jack Lloyd
+ * (C) 2024 René Meusel, Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ */
+
+#ifndef BOTAN_KYBER_HELPERS_H_
+#define BOTAN_KYBER_HELPERS_H_
+
+#include <botan/internal/ct_utils.h>
+#include <botan/internal/kyber_constants.h>
+#include <botan/internal/loadstor.h>
+#include <botan/internal/pqcrystals_helpers.h>
+
+namespace Botan {
+
+/**
+ * Special load_le<> that takes 3 bytes and returns a 32-bit integer.
+ */
+inline uint32_t load_le3(std::span<const uint8_t, 3> in) {
+   return Botan::load_le(std::array<uint8_t, 4>{in[0], in[1], in[2], 0});
+}
+
+/**
+ * NIST FIPS 203 IPD, Formula 4.5 (Compress)
+ */
+template <size_t d>
+   requires(d > 0 && d < 12)
+constexpr std::make_unsigned_t<KyberConstants::T> kyber_compress(KyberConstants::T x) {
```

Comment:
Seems like a decent case for `auto` return type imo

Reply:
Mhh, yes and no. I feel there's some value in having the explicit return type for methods that are used elsewhere (and where the type isn't overly excessive).